sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -50,28 +50,44 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
   override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
     case AddColumns(ResolvedV1TableIdentifier(ident), cols) =>
       cols.foreach { c =>
-        assertTopLevelColumn(c.name, "AlterTableAddColumnsCommand")
+        if (c.name.length > 1) {
+          throw QueryCompilationErrors.operationOnlySupportedWithV2TableError(
+            ident.catalog.getOrElse(""),
+            ident.database.getOrElse(""),
```

Comment:
`ident` must be qualified, we can do `ident.catalog.get` and `ident.database.get`

Reply:
Fixed. Thanks